File: Python/01.ProgrammingBasics/06.Conditional-Statements-Adv-Exercise/03.NewHouse/_03.NewHouse.py
flower_type = input()
flower_count = int(input())
budget = int(input())

flower_prices = {
    "Roses": 5.00,
    "Dahlias": 3.80,
    "Tulips": 2.80,
    "Narcissus": 3.00,
    "Gladiolus": 2.50
}

total_cost = flower_count * flower_prices[flower_type]

if flower_type == "Roses" and flower_count > 80:
    total_cost *= 0.90
elif flower_type == "Dahlias" and flower_count > 90:
    total_cost *= 0.85
elif flower_type == "Tulips" and flower_count > 80:
    total_cost *= 0.85
elif flower_type == "Narcissus" and flower_count < 120:
    total_cost *= 1.15
elif flower_type == "Gladiolus" and flower_count < 80:
    total_cost *= 1.20

if budget >= total_cost:
    left_money = budget - total_cost
    print(f"Hey, you have a great garden with {flower_count} {flower_type} and {left_money:.2f} leva left.")
else:
    needed_money = total_cost - budget
    print(f"Not enough money, you need {needed_money:.2f} leva more.")

File: C++/01-Basics/03.3-ConditionalStatementsAdvЕxercise/03.NewHouse/03.NewHouse.cpp
#include <iostream>
#include <string>
#include <iomanip>

using namespace std;

int main() {
    string flowerType;
    int flowerCount;
    int budget;
    cin >> flowerType >> flowerCount >> budget;

    double price = 0.0;

    if (flowerType == "Roses") {
        price = 5.00 * flowerCount;
        if (flowerCount > 80) {
            price *= 0.90;  // 10% discount
        }
    }
    else if (flowerType == "Dahlias") {
        price = 3.80 * flowerCount;
        if (flowerCount > 90) {
            price *= 0.85;  // 15% discount
        }
    }
    else if (flowerType == "Tulips") {
        price = 2.80 * flowerCount;
        if (flowerCount > 80) {
            price *= 0.85;  // 15% discount
        }
    }
    else if (flowerType == "Narcissus") {
        price = 3.00 * flowerCount;
        if (flowerCount < 120) {
            price *= 1.15;  // 15% more expensive
        }
    }
    else if (flowerType == "Gladiolus") {
        price = 2.50 * flowerCount;
        if (flowerCount < 80) {
            price *= 1.20;  // 20% more expensive
        }
    }

    double diff = abs(budget - price);

    if (budget >= price) {
        cout << fixed << setprecision(2) << "Hey, you have a great garden with " << flowerCount << " " << flowerType
            << " and " << diff << " leva left." << endl;
    }
    else {
        cout << fixed << setprecision(2) << "Not enough money, you need " << diff << " leva more." << endl;
    }

    return 0;
}
